src/NClient/NClient.Abstractions/INClientBuilder.cs

```diff
@@ -16,5 +17,11 @@ INClientApiBuilder<TClient> For<TClient>(string host)
         /// <typeparam name="TClient">The type of interface of controller used to create the client.</typeparam>
         INClientApiBuilder<TClient> For<TClient>(Uri host)
             where TClient : class;
+        
+        /// <summary>Sets the main client settings.</summary>
+        /// <param name="host">The object with base address of URI used when sending requests.</param>
+        /// <typeparam name="TClient">The type of interface of controller used to create the client.</typeparam>
+        INClientApiBuilder<TClient> For<TClient>(IHost? host)
```

Comment:
Why is the host optional?

Reply:
We can use both option:
1) in For method
2) in With method

So.. neither could be required, it's build-time verification.

Reply:
Then creating a client will look like this:
```C#
...
    .For<IClient>(null)
    .WithHost("https://...")
....
```
It seems this is not what users want :) I suggest making the host non-optional and adding the method `INCluentApiBuilder<TClient> For<TClient>()`

Reply:
Ok. No more "WithHost" methods :-1: